apis/python/src/tiledbsoma/_arrow_types.py

```diff
@@ -159,3 +160,31 @@ def tiledb_schema_to_arrow(tdb_schema: tiledb.ArraySchema) -> pa.Schema:
         arrow_schema_dict[name] = arrow_type_from_tiledb_dtype(attr.dtype, attr.isascii)
 
     return pa.schema(arrow_schema_dict)
+
+
+def df_to_arrow(df: pd.DataFrame) -> pa.Table:
+    """
+    Categoricals are not yet well supported, so we must flatten.
+    Also replace Numpy/Pandas-style nulls with Arrow-style nulls.
+    """
+    null_fields = set()
+    for k in df:
```

Comment:
I see this is just moved code, so you don’t need to make this change if you don’t want to, but:

Dataframes implement the `dict`-like `.items` method, so this can be `for name, col in df.items()` to make some usage of `df[k]` a bit more readable

Reply:
@thetorpedodog no b/c I want `df[k]` as left-hand sides to mutate